src/lib/storage/value_segment.cpp

```diff
@@ -6,68 +6,84 @@
 namespace opossum {
 
 template <typename T>
-ValueSegment<T>::ValueSegment(bool nullable) {
-  // Implementation goes here
-}
+ValueSegment<T>::ValueSegment(bool nullable) : _is_nullable(nullable) {}
 
 template <typename T>
 AllTypeVariant ValueSegment<T>::operator[](const ChunkOffset chunk_offset) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (is_null(chunk_offset)) {
+    return NULL_VALUE;
+  }
+
+  return _values.at(chunk_offset);
 }
 
 template <typename T>
 bool ValueSegment<T>::is_null(const ChunkOffset chunk_offset) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (!is_nullable()) {
+    return false;
+  }
+
+  return _nulls.at(chunk_offset);
 }
 
 template <typename T>
 T ValueSegment<T>::get(const ChunkOffset chunk_offset) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (is_null(chunk_offset)) {
+    Fail("No value present at offset");
+  }
```

Comment:
How come your are using `Fail` here? A more suitable representation might be:

```suggestion
  DebugAssert(!is_null(chunk_offset), "No value present at offset");
```

Keep in mind, that `ChunkOffset` is an alternative naming for `uint32_t` (see types.hpp).

Reply:
Sure, using an assertion makes sense here. I don't get your comment on the `ChunkOffset` though

Reply:
If you are aware of it, than it does not play a role at all.